_episodes/01-why-bother-with-performance.md

```diff
@@ -0,0 +1,255 @@
+> inside the smaller box. If you wanted to simulate what was happening inside the large
+> box for 8 hours, how long will the example below take to run if we only use one CPU
+> core?
+>
+> ![10x10x10_cube](../fig/01/10x10x10_cube.png)
+>
+> {: language-bash}
+>
+>> ## Solution
+>>
+>> 8000 hours...close to a year!
+>>
+>> This is way longer than anyone could bear! But remember, that is utilising just one
+>> core. If you had a machine that could simulate each of those smaller boxes
+>> simultaneously and a code that enables each box to effectively interact with each
+>> other, the whole job would only take roughly an hour (but probably a little more).
+>>
+>{: .solution}
+{: .challenge}
+
+## How is performance measured?
+
+There are a number of key terms in computing when it comes to understanding performance
+and expected duration of tasks. The most important of these are **walltime**, **flops**
+and **CPU hours**.
+
+> ## Walltime
+>
+> Walltime is simply the length of time, usually measured in seconds, that a program
+> takes to run (i.e., to execute its assigned tasks). It is not directly related to the
+> resources used, it's simply the time it takes according to an independent clock on the
+> wall.
+>
+{: .callout}
+
+> ## Flops
+>
+> Flops (of flop/s) stands for floating point operations per second and they are
+> typically used to measure the (theoretical) performance of a computer's processor.
+>
+> The theoretical peak flops is given by
+>
+> `Number of cores * Average frequency * Operations per cycle`
+>
+> What a software program can *achieve* in terms of flops is usually a surprisingly
+> small percentage of this value (e.g., 10% efficiency is not a bad number!).
+>
+> Since in our case we don't develop or compile the code, the only influence we have
+> on the flops achieved by the application is by dictating the choices it makes during
+> execution (sometimes called telling it what *code path* to walk).
+>
+{: .callout}
+
+> ## Calculate Flops
+>
+> Can you find how many flops your computer is capable of at maximum efficiency?
+>
+{: .challenge}
+
+> ## CPU Hours
+>
+> CPU hours is the amount of CPU time spent processing. For example, if I execute for a
+> walltime of 1 hour on 10 CPUs, then I will have used up 10 CPU hours.
+>
+> Maximising the flops of our application will help reduce the CPU hours, since we
+> will squeeze more calculations out of the same CPU time. However, you can achieve much
+> greater influence on the CPU hours by using a better algorithm to get to your result.
+> The best algorithm choice is usually heavily dependent on your particular use case
+> (and you are always limited by what algorithm options are available in the software).
+>
+{: .callout}
+
+
+> ## Calculate CPU hours
+>
+> In the following example, assume that you are utilising all the available core in a
+> node. Calculate the CPU hours requested to run the following job. Each node has 40
```

Comment:
Let's put the core count inside the template so we don't have to explicitly mention it (it's going to be different on every system)